helix-rest/src/main/java/org/apache/helix/rest/clusterMaintenanceService/MaintenanceManagementService.java

```diff
@@ -81,43 +82,65 @@ public class MaintenanceManagementService {
   public static final String HELIX_CUSTOM_STOPPABLE_CHECK = "CustomInstanceStoppableCheck";
   public static final String OPERATION_CONFIG_SHARED_INPUT = "OperationConfigSharedInput";
 
+  public static final Set<StoppableCheck.Category> SKIPPABLE_HEALTH_CHECK_CATEGORIES =
+      ImmutableSet.of(StoppableCheck.Category.CUSTOM_INSTANCE_CHECK,
+          StoppableCheck.Category.CUSTOM_PARTITION_CHECK);
+
   private final ConfigAccessor _configAccessor;
   private final CustomRestClient _customRestClient;
   private final String _namespace;
   private final boolean _skipZKRead;
   private final HelixDataAccessorWrapper _dataAccessor;
   private final Set<String> _nonBlockingHealthChecks;
+  private final Set<StoppableCheck.Category> _skipHealthCheckCategories;
 
   public MaintenanceManagementService(ZKHelixDataAccessor dataAccessor,
       ConfigAccessor configAccessor, boolean skipZKRead, String namespace) {
-    this(dataAccessor, configAccessor, CustomRestClientFactory.get(), skipZKRead,
+    this(new HelixDataAccessorWrapper(dataAccessor, CustomRestClientFactory.get(), namespace),
```

Comment:
can you please clarify why are we changing each method to pass wrapper where as the earlier implementation had just one implementation which converted HelixDataAccessor to HelixDataAccessorWrapper?

Reply:
This makes it possible to mock HelixDataAccessorWrapper in tests. If you create the object in the final constructor, you cannot mock it.

Reply:
this makes sense. thanks